libvmi/driver/xen/xen_events.c

```diff
@@ -1878,8 +1993,11 @@ static
 status_t wait_for_event_or_timeout(vmi_instance_t vmi, unsigned long ms, bool *needs_unmasking)
 {
     xen_events_t *xe = xen_get_events(vmi);
-
-    switch ( poll(&xe->fd, 1, ms) ) {
+#ifdef HAVE_LIBXENSTORE
+    switch ( poll(xe->fd, 2, ms) ) {
```

Comment:
Just because xenstore is enabled it doesn't mean domain watch is. Only poll on it if domain watch is explicitely enabled.